debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresSnapshotChangeEventSource.java

```diff
@@ -260,11 +261,25 @@ protected SchemaChangeEvent getCreateTableEvent(RelationalSnapshotContext<Postgr
 
     @Override
     protected void completed(SnapshotContext<PostgresPartition, PostgresOffsetContext> snapshotContext) {
+        try {
+            jdbcConnection.commit();
+        } catch (SQLException sqle) {
+            LOGGER.error("Exception while committing prior to reporting snapshot completion {}", sqle);
+            throw new DebeziumException(sqle);
+        }
+
         snapshotter.snapshotCompleted();
     }
 
     @Override
     protected void aborted(SnapshotContext<PostgresPartition, PostgresOffsetContext> snapshotContext) {
+        try {
```

Comment:
When do we call aborted? Should we all .rollback?

Reply:
aborted is called when there's any connector restart or task restart when snapshot is in progress or if there's any `SQLException` during snapshot. From JDBC connection perspective, I think it doesn't matter much whether we commit or rollback because we only execute a `SELECT *` query in that transaction, please correct me if I am misunderstanding.

However, I have made the change to call `rollback()` now when the snapshot is aborted.